gammapy/datasets/map.py

```diff
@@ -1671,21 +1672,22 @@ def __init__(
         gti=None,
         evaluation_mode="local",
     ):
+
         self.model = model
         self.exposure = exposure
         self.psf = psf
         self.edisp = edisp
         self.gti = gti
         self.contributes = True
+        self._npred = None
```

Comment:
Maybe rename to `self._npred_cached` and `self._pars_cached` to be a bit more explicit?